### ISSUE TITLE
net: NET_CONTEXT_SYNC_RECV relevant

### ISSUE THREAD

**Issue by zephyrbot** [zephyrproject-rtos/zephyr#4060]
**_Reported by Likai Liu:_**

Find in v1.8.0

Thread occupying CPU forever(if no_preempt).

Set NET_CONTEXT_SYNC_RECV y

If calls net_context_recv more than once use a same context, the thread will occupy CPU forever.

in code:

```
while (k_sem_take(&context->recv_data_wait, K_NO_WAIT)) {
			;
		}
```
		
The issue can be reappeared easily by using shell command  “zperf> udp.upload 10.237.164.178 1111 1 1K 1M”

@jukkar jukka :)

(Imported from Jira ZEP-2637)

**Comment by jukkar** [comment]
Should be fixed by #4571

**Comment by nashif** [comment]
fixed, so close it.
